ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroDeserializerTest.java

```diff
@@ -201,4 +213,342 @@ public void shouldDeserializeWithMissingFields() {
 
     return kafkaAvroSerializer.serialize(topicName, avroRecord);
   }
+
+  private void shouldDeserializeTypeCorrectly(final org.apache.avro.Schema avroSchema,
+                                              final Object avroValue,
+                                              final Schema ksqlSchema) {
+    shouldDeserializeTypeCorrectly(avroSchema, avroValue, ksqlSchema, avroValue);
+  }
+
+  private void shouldDeserializeTypeCorrectly(final org.apache.avro.Schema avroSchema,
+                                              final Object avroValue,
+                                              final Schema ksqlSchema,
+                                              final Object ksqlValue) {
+    final SchemaRegistryClient schemaRegistryClient = new MockSchemaRegistryClient();
+
+    final org.apache.avro.Schema avroRecordSchema = org.apache.avro.SchemaBuilder.record("test_row")
+        .fields()
+        .name("field0")
+        .type(avroSchema)
+        .noDefault()
+        .endRecord();
+    final Schema ksqlRecordSchema = SchemaBuilder.struct().field("field0", ksqlSchema).build();
+
+    final GenericRecord avroRecord = new GenericData.Record(avroRecordSchema);
+    avroRecord.put("field0", avroValue);
+
+    final Map map = new HashMap();
```

Comment:
nit: `Map<String, Object>`

Reply:
Also some overlap here with the same in `getSerializedRow`. Perhaps extract a method to create the `GenericRecord` ?